sql/server/src/main/java/io/stargate/api/sql/server/avatica/StatementHolder.java

```diff
@@ -52,12 +58,25 @@ public static StatementHolder empty(String connectionId, int id, DataStore dataS
     return new StatementHolder(connectionId, id, new Prepared(null, null), dataStore);
   }
 
-  public Prepared prepared() {
-    return prepared;
+  public ExecutionResult result() {
+    return result;
   }
 
-  public Prepared prepare(String sql) {
-    return prepare(sql, dataStore);
+  private ExecutionResult execute(Prepared prepared, List<TypedValue> params) {
+    List<Object> localParams =
+        params.stream().map(TypedValue::toLocal).collect(Collectors.toList());
+    Iterable<Object> rows = prepared.query().execute(localParams);
+    ExecutionResult r = new ExecutionResult(prepared, rows.iterator());
+    result = r;
+    return r;
```

Comment:
Minor but can this be

```suggestion
    result = new ExecutionResult(prepared, rows.iterator());
    return result;
```